R/ChromBackend.R

```diff
@@ -0,0 +1,615 @@
+#'   `mzMax` returns a `numeric` of length equal to the number of chromatograms
+#'   in `object`, `mzMax` expects a `numeric` of length equal to the number
+#'   of chromatograms in `object`.
+#'
+#' - `mzMin`,`mzMin<-`: gets or sets the lower m/z of the mass-to-charge range
+#'   from which the chromatogram contains signal (e.g. if the chromatogram
+#'   was extracted from MS data in spectra format and a m/z range was provided).
+#'   `mzMin` returns a `numeric` of length equal to the number of chromatograms
+#'   in `object`, `mzMin` expects a `numeric` of length equal to the number
+#'   of chromatograms in `object`.
+#'
+#' - `pairs`: returns a `list` of matrices with the data pairs from each
+#'   chromatogram. The length of the `list` is equal to the number of
+#'   chromatograms in `object`. Each element is a two-column `numeric` `matrix`
+#'   with the retention time (first column) and intensity values (second column)
+#'   of one chromatogram. For an empty chromatogram a `matrix` with 0 rows and
+#'   two columns (named `"rtime"` and `"intensity"`) has to be returned.
+#'
+#' - `pairs<-`: replaces the chromatogram data of the backend. This function
+#'   expects a `list` of two-column matrices in the format returned by `pairs`.
+#'   The length of `value` has to be identical to the number of chromatograms
+#'   in `object`. Note that just writeable backends support this method.
+#'
+#' - `precursorMz`,`precursorMz<-`: gets or sets the (target) m/z of the
+#'   precursor (for SRM data). `precursorMz` returns a `numeric` of length
+#'   equal to the number of chromatograms in `object`. `precursorMz<-` expects
+#'   a `numeric` of length equal to the number of chromatograms.
+#'
+#' - `productMz`,`productMz<-`: gets or sets the (target) m/z of the
+#'   product (for SRM data). `productMz` returns a `numeric` of length
+#'   equal to the number of chromatograms in `object`. `productMz<-` expects
+#'   a `numeric` of length equal to the number of chromatograms.
+#'
+#' - `precursorMzMin`,`precursorMzMax`,`productMzMin`, `productMzMax`: get
+#'   the lower and upper margin for the precursor or product isolation windows.
+#'
+#' - `rtime`: gets the retention times from the chromatograms. returns a
+#'   [NumericList()] of `numeric` vectors (retention times for each
+#'   chromatogram). The length of the returned list is equal to the number of
+#'   chromatograms in `object`.
+#'
+#' - `rtime<-`: replaces the retention times. `value` has to be a `list` (or
+#'   [NumericList()]) of length equal to the number of chromatograms and the
+#'   number of values within each list element identical to the number of
+#'   data pairs in each chromatogram. Note that just writeable backends support
+#'   this method.
+#'
+#' - `selectChromVariables`: reduce `object` retaining only specified
+#'   chromatogram variables.
+#'
+#' @author Johannes Rainer, Sebastian Gibb, Laurent Gatto
+#'
+#' @md
```

Comment:
BTW: we could add `Roxygen: list(markdown=TRUE)` to `DESCRIPTION` (as I did in `MsCoreUtils`). I think we will write all roxygen documentation in markdown. So we can avoid adding `@md` to each roxygen comment.

Reply:
Good point. I will add that.